Packet++/src/Sll2Layer.cpp

```diff
@@ -0,0 +1,134 @@
+#define LOG_MODULE PacketLogModuleSll2Layer
+
+#include "Sll2Layer.h"
+#include "Logger.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "PayloadLayer.h"
+#include "ArpLayer.h"
+#include "VlanLayer.h"
+#include "PPPoELayer.h"
+#include "MplsLayer.h"
+#include <string.h>
+#include "EndianPortable.h"
+
+namespace pcpp
+{
+Sll2Layer::Sll2Layer(uint32_t interfaceIndexType, uint16_t ARPHRDType, uint8_t packetType)
+{
+	const size_t headerLen = sizeof(sll2_header);
+	m_DataLen = headerLen;
+	m_Data = new uint8_t[headerLen];
+	memset(m_Data, 0, headerLen);
+	sll2_header* sll2Hdr = (sll2_header*)m_Data;
+	sll2Hdr->packet_type = packetType;
+	sll2Hdr->ARPHRD_type = htobe16(ARPHRDType);
+	sll2Hdr->interface_index_type = htobe32(interfaceIndexType);
+	m_Protocol = SLL2;
+}
+
+bool Sll2Layer::setLinkLayerAddr(uint8_t* addr, size_t addrLength)
+{
+	if (addrLength == 0 || addrLength > 8)
```

Comment:
we can add here a check that `addr != nullptr`

Reply:
ok,I also add it in SllLayer.cpp